numba/_pymodule.h

```diff
@@ -24,5 +24,10 @@
 #define PyInt_Type PyLong_Type
 #define PyInt_Check PyLong_Check
 #define PyInt_CheckExact PyLong_CheckExact
+// Define object attribute using a temporary variable to avoid memory leak
+#define SetAttrStringFromVoidPointer(m, name, tmp) \
```

Comment:
I think you had better wrap these three lines with a `do { } while (false)` block and declare a temporary variable in the block instead of using an external variable.

Reply:
This is now applied, but I used `while (0)` instead of `while (false)` because `false` is not defined everywhere the macro is used.